doc/source/structures/misc/lexicon.rst

```diff
@@ -0,0 +1,160 @@
+.. _lexicon:
+
+Lexicon
+====
+
+A :struct:`Lexicon` is an associative array where the keys and values can be of any type in kOS. You can create your own :struct:`Lexicon` variables and kOS itself can return them to you. 
```

Comment:
About this -> "and kOS itself can return them to you" :
I had a read through the code and couldn't find where we return one of these from any method or function call.  Is this sentence here for the sake of future expansion?  (i.e. That you anticipate that we'll start returning associative arrays from the system later, but that's not in this particular PR?)